middlewares/userAuth.js

```diff
@@ -32,9 +36,18 @@ const userAuth = {
     });
   },
 
-  isAuthorized: (req, res, next) => {
-    // TO DO: check if user is authorized
-    next();
+  isAuthorized: async (req, res, next) => {
+    const commentRoute = '/api/v1/comments';
+    if (req.baseUrl === commentRoute) {
+      // Compare the comment user_id to the user making the request
+      const comment = await CommentModel.findOne({ _id: req.params.id }, { user_id: 1, _id: 0 });
+      const user = await UserModel.findOne({ username: req.authUser.username }, { _id: 1 });
```

Comment:
cool!! for line 44, probably can put outside of the `if (req.baseUrl ===...) ` , since it's very likely to be re-used by other routes.

will try on my route once this is merged!

Reply:
Good point!